res/controllers/Traktor-Kontrol-S2-MK1-hid-scripts.js

```diff
@@ -0,0 +1,1361 @@
+    if (field.value > 0) {
+        var effectUnitNumber = field.group.slice(-2, -1);
+        if (TraktorS2MK1.shiftPressed["[Channel" + effectUnitNumber + "]"]) {
+            engine.setValue(field.group, "load_preset", 1);
+            return;
+        }
+        TraktorS2MK1.effectFocusLongPressTimer[field.group] = engine.beginTimer(TraktorS2MK1.longPressTimeoutMilliseconds, function() {
+            TraktorS2MK1.effectFocusChooseModeActive[field.group] = true;
+            TraktorS2MK1.effectButtonLEDconnections[field.group].forEach(function(connection) {
+                connection.disconnect();
+            });
+            var makeButtonLEDcallback = function(buttonNumber) {
+                return function(value, group, _control) {
+                    TraktorS2MK1.controller.setOutput(group, "!effectbutton" + buttonNumber,
+                        value === buttonNumber ? ButtonBrightnessOn : ButtonBrightnessOff, !TraktorS2MK1.batchingLEDUpdate);
+                };
+            };
+            TraktorS2MK1.batchingLEDUpdate = true;
+            for (var i = 0; i <= 2; i++) {
+                TraktorS2MK1.effectButtonLEDconnections[i] = engine.makeConnection(
+                    field.group, "focused_effect", makeButtonLEDcallback(i+1));
+                TraktorS2MK1.effectButtonLEDconnections[i].trigger();
+            }
+            TraktorS2MK1.batchingLEDUpdate = false;
+            TraktorS2MK1.effectButtonLEDconnections[2].trigger();
+        });
+        if (!showParameters) {
+            engine.setValue(field.group, "show_parameters", 1);
+            TraktorS2MK1.effectFocusButtonPressedWhenParametersHidden[field.group] = true;
+        }
+    } else {
+        if (TraktorS2MK1.effectFocusLongPressTimer[field.group] !== 0) {
+            engine.stopTimer(TraktorS2MK1.effectFocusLongPressTimer[field.group]);
+            TraktorS2MK1.effectFocusLongPressTimer[field.group] = 0;
+        }
+
+        if (TraktorS2MK1.effectFocusChooseModeActive[field.group]) {
+            TraktorS2MK1.effectFocusChooseModeActive[field.group] = false;
+            TraktorS2MK1.connectEffectButtonLEDs(field.group);
+        } else if (showParameters && !TraktorS2MK1.effectFocusButtonPressedWhenParametersHidden[field.group]) {
+            engine.setValue(this.group, "show_parameters", 0);
+        }
+
+        TraktorS2MK1.effectFocusButtonPressedWhenParametersHidden[field.group] = false;
+    }
+};
+
+// Refer to https://github.com/mixxxdj/mixxx/wiki/standard-effects-mapping for how to use this.
+TraktorS2MK1.effectKnob = function(field) {
+    var knobNumber = parseInt(field.id.slice(-1));
+    var effectUnitGroup = field.group;
+    var focusedEffect = engine.getValue(effectUnitGroup, "focused_effect");
+    if (focusedEffect > 0) {
+        engine.setParameter(effectUnitGroup.slice(0, -1) + "_Effect" + focusedEffect + "]",
+            "parameter" + knobNumber,
+            field.value / 4096);
```

Comment:
It does not matter. The middle position is typically of by about 10 and it is really difficult to get the value `0x7ff` values do not increment by 1. This is probably some sort of filter to smooth out the ADC readings and not have to send a lot of packets because of the noise.